app/web/src/newhotness/logic_composables/search.ts

```diff
@@ -33,8 +41,36 @@ export function useComponentSearch(
   searchString: MaybeRefOrGetter<string | undefined>,
   componentsRef: MaybeRefOrGetter<ComponentInList[] | undefined>,
 ): Ref<ComponentInList[] | undefined> {
-  // This listens for changes to attributes so the search will be re-run if they change.
-  const queryAttributes = useQueryAttributes();
+  const queryClient = useQueryClient();
+
+  const startTerms = reactive<QueryAttributesTerm[]>([]);
+  const exactTerms = reactive<QueryAttributesTerm[]>([]);
+  const terms = computed(() => [
+    ...startTerms.map((t) => toRaw(t)),
+    ...exactTerms.map((t) => toRaw(t)),
+  ]);
+  const termsAsId = computed(() => terms.value.join("/"));
+
+  const ctx = useContext();
+  const makeKey = useMakeKey();
+  const queryKey = makeKey(EntityKind.QueryAttributes, termsAsId);
+  // Use tanstack `useQuery` so we can bust the cache of QueryAttributes whenever an
+  // AttributeTree is updated. We don't actuall;y do the query in here though, we just return
+  // a function that *can* do the query!
+  const attrQuery = useQuery({
+    queryKey,
+    staleTime: 0,
+    queryFn: async () =>
+      await bifrostQueryAttributes(
+        ctx.workspacePk.value,
+        ctx.changeSetId.value,
+        terms.value,
+      ),
+  });
+
+  const componentIds = computed<Set<string>>(() => {
```

Comment:
note: <Set<string>> isn't strictly needed here, it's implied by the new Set(). Could just be:

```
  const componentIds = computed(() => new Set(attrQuery.data.value ?? []));
```